fastlane/Fastfile

```diff
@@ -86,6 +86,41 @@ platform :ios do
     Spaceship::Tunes::SandboxTester.create!(email: email, password: password)
   end
 
+  desc "Deployment checks"
+  lane :deployment_checks do |options|
+    version_number = get_version_number(target: "Purchases")
+    check_no_git_tag_exists(version_number)
+    check_pods
+    carthage_archive
+    check_no_github_release_exists(version_number)
+  end
+
+end
+
+def check_no_git_tag_exists(version_number)
+  if git_tag_exists(tag: version_number, remote: true, remote_name: 'origin')
+    raise "git tag with version #{version_number} already exists!"
+  end
+end
+
+
+def check_no_github_release_exists(version_number)
+  found_release_number = get_github_release(url: "revenuecat/purchases-ios", version: version_number)
+  raise "Release with version #{version_number} already exists!" unless found_release_number.nil?
+end
+
+def carthage_archive
+  # can't use fastlane carthage integration directly because of Carthage/Xcode 12 compatibility issues:
+  # https://github.com/Carthage/Carthage/issues/3019
```

Comment:
This makes me so sad everytime I see it. I hope they fix it soon haha

Reply:
+1